orchagent/switchorch.cpp

```diff
@@ -0,0 +1,85 @@
+#include <map>
+
+#include "switchorch.h"
+
+using namespace std;
+using namespace swss;
+
+extern sai_object_id_t gSwitchId;
+extern sai_switch_api_t *sai_switch_api;
+
+const map<string, sai_switch_attr_t> switch_attribute_map =
+{
+    {"fdb_unicast_miss_packet_action",      SAI_SWITCH_ATTR_FDB_UNICAST_MISS_PACKET_ACTION},
+    {"fdb_broadcast_miss_packet_action",    SAI_SWITCH_ATTR_FDB_BROADCAST_MISS_PACKET_ACTION},
+    {"fdb_multicast_miss_packet_action",    SAI_SWITCH_ATTR_FDB_MULTICAST_MISS_PACKET_ACTION}
+};
+
+const map<string, sai_packet_action_t> packet_action_map =
+{
+    {"drop",    SAI_PACKET_ACTION_DROP},
+    {"forward", SAI_PACKET_ACTION_FORWARD},
+    {"trap",    SAI_PACKET_ACTION_TRAP}
+};
+
+SwitchOrch::SwitchOrch(DBConnector *db, string tableName) :
+        Orch(db, tableName)
+{
+}
+
+void SwitchOrch::doTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        auto t = it->second;
+
+        auto op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            for (auto i : kfvFieldsValues(t))
+            {
+                auto attribute = fvField(i);
+
+                if (switch_attribute_map.find(attribute) == switch_attribute_map.end())
+                {
+                    SWSS_LOG_ERROR("Unsupported switch attribute %s", attribute.c_str());
+                    it = consumer.m_toSync.erase(it);
+                    continue;
+                }
+
+                auto value = fvValue(i);
+                if (packet_action_map.find(value) == packet_action_map.end())
```

Comment:
It is not so generic here. What if I want to change a switch attribute whose value is not about pack_action?

Reply:
good point. I think we could patch that in a more general way. thanks for the feedback.